documentation/tig_stack.md

```diff
@@ -0,0 +1,340 @@
+project from the `tig_stack` directory. The project has the following file structure:
+
+```shell
+.
+├── Puppetfile
+├── Vagrantfile
+├── bolt-project.yaml
+├── hiera.yaml
+├── inventory.yaml
+├── manifests
+│   ├── dashboard.pp
+│   ├── params.pp
+│   └── telegraf.pp
+├── plans
+│   └── init.pp
+└── templates
+    └── dashboards
+        └── telegraf.json
+```
+
+## Provision your targets
+
+A typical real-world environment would consist of multiple agents, but for the
+purposes of this guide, you'll provision two virtual machines to use as
+targets - `target0` and `target1`. The first target hosts the Garafana dashboard
+and the InfluxDB database as well as a Telegraf agent. The second target hosts
+only a Telegraf agent. If you'd prefer to use your own existing targets, skip
+this section.
+
+The `bolt_tig` directory contains the following `Vagrantfile`:
+
+```shell
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+TARGETS = 2
+
+Vagrant.configure(TARGETS) do |config|
+  config.vm.box = "centos/7"
+  config.ssh.forward_agent = true
+
+  TARGETS.times do |i|
+    config.vm.define "target#{i}" do |target|
+      target.vm.hostname = "target#{i}"
+      target.vm.network :private_network, ip: "10.0.0.#{100 + i}"
+    end
+  end
+end
+```
+
+To provision your targets:
+1. Generate the SSH configuration for both targets. Bolt will automatically
+   detect the configuration.
+
+    ```shell
+    mkdir ~/.ssh
+    vagrant ssh-config | sed /StrictHostKeyChecking/d | sed /UserKnownHostsFile/d >> ~/.ssh/config
+    ```
+2. Make sure you can SSH into the targets. For example:
+   ```shell
+   ssh vagrant@target0
+   ```
+Next, install the Puppet modules for the different components of the TIG stack. 
+
+## Install the TIG modules from a Puppetfile
+
+Before you can use Bolt to install modules, you must create a Puppetfile. A
+Puppetfile is a formatted text file that contains a list of modules and their
+versions. It can include modules from the Puppet Forge or a git repository.
+
+The `bolt_tig` directory contains the following `Puppetfile`:
+```puppet
+mod 'puppet-grafana', '6.0.0'
+mod 'quadriq-influxdb', '0.2.1'
+mod 'puppetlabs-stdlib', '5.2.0'
+mod 'puppet-telegraf', '2.1.0'
+```
+
+To install the modules for the TIG stack:
+1. Install the modules using the following command:
+    ```shell
+    bolt puppetfile install
+    ```
+
+2. Install the `toml-rb` gem, which is required by the `puppet-telegraf` module:
+    ```shell
+    /opt/puppetlabs/bolt/bin/gem install toml-rb
+    ```
+
+Next, create an inventory file to specify which targets to use as part of the plan.
+
+## Create an inventory file
+
+Use an inventory file to group and configure the connection settings for your
+targets. If you're using existing targets in your system, replace `target0` and
+`target1` with your own targets.
+
+The `bolt_tig` directory contains the following inventory file:
+
+```yaml
+version: 2
+groups:
+  - name: dashboard
+    targets: [ target0 ]
```

Comment:
```suggestion
    targets:
      - target0
```

I think we show arrays of targets like this elsewhere in the docs. This would also be updated in the tig repo.